batcher/aligned/src/main.rs

```diff
@@ -107,20 +109,51 @@ pub struct SubmitArgs {
     keystore_path: Option<PathBuf>,
     #[arg(name = "Private key", long = "private_key")]
     private_key: Option<String>,
+    #[arg(name = "Max Fee", long = "max_fee")]
+    max_fee: Option<String>, // String because U256 expects hex
     #[arg(
-        name = "Max Fee",
-        long = "max_fee",
-        default_value = "1300000000000000" // 13_000 gas per proof * 100 gwei gas price (upper bound)
+        name = "Price Estimate",
+        long = "price_estimate",
+        default_value = "default",
+        help = "`price_estimate` allows for a user to use the aligned_sdk `estimate_fee` function to compute there `max_fee`.\n Usage: `default,`, `instant`, \"custom <NUM_PROOFS_IN_BATCH>\"",
+        allow_hyphen_values = true
     )]
-    max_fee: String, // String because U256 expects hex
+    price_estimate: Option<String>,
     #[arg(name = "Nonce", long = "nonce")]
     nonce: Option<String>, // String because U256 expects hex
     #[arg(
         name = "The working network's name",
         long = "network",
         default_value = "devnet"
     )]
-    network: NetworkArg,
+    network: Network,
+}
+
+impl SubmitArgs {
+    async fn get_max_fee(&self) -> Result<U256, AlignedError> {
+        // If max_fee is explicitly provided it is used first
+        if let Some(max_fee) = &self.max_fee {
+            // Inform the user if both are declared that `max_fee` is used.
+            if self.price_estimate.is_some() {
+                warn!("`max_fee` and `price_estimate` are both present using `max_fee`");
+            }
```

Comment:
I'd prefer no to make decisions for the users. I'd return here and tell the user to provide one or the other.

Reply:
I agree its a better patten!